test/Microsoft.AspNetCore.TestHost.Tests/TestClientTests.cs

```diff
@@ -239,6 +239,54 @@ public void Dispose()
             clientSocket.Dispose();
         }
 
+        [ConditionalFact]
+        [FrameworkSkipCondition(RuntimeFrameworks.Mono, SkipReason = "Hangs randomly (issue #507)")]
```

Comment:
Sure we can do a clean up, but there's no point in adding more code we are going to remove later.